column_computation.hh

```diff
@@ -36,7 +36,7 @@ public:
     virtual column_computation_ptr clone() const = 0;
 
     virtual bytes serialize() const = 0;
-    virtual bytes_opt compute_value(const schema& schema, const partition_key& key, const clustering_row& row) const = 0;
+    virtual bytes compute_value(const schema& schema, const partition_key& key) const = 0;
```

Comment:
I mentioned before that general computed column can and do depend on the row, and can return null (for example, if an input they depend on is null). I suppose we can undo this when we implemented general computed columns.

Reply:
My first comment on this patch was also that I don't understand why do this change it all. If the "row" parameter was unused, we could have kept it unused and not fix it today. 

But I agree with @enedil that if the new code does not use the row key (it does use the row and get it from some other mechanism!) and doesn't return null ever (unless I'm missing something?) then new API of this function is fine, and we don't to anticipate the future - we can always change it again later when the need arises.

We just need to make a decision here.

Reply:
I'm fine with leaving the original `bytes_opt` return value, it makes this already big series a little bit smaller.

Reply:
I don't know what to do about this. Both ways (leave it bytes_opt as it was in the past, or leave it bytes as it is now in this patch) make some sense, but leaving it as it is **now** (bytes) is a bit easier for me because I don't need to change anything...
Another reason why I think it makes sense to leave it as it is now (bytes), is that the old code checked if the opt_bytes was null, and if it was, did on_internal_error. So maybe it's better just not to have this option?